libcrux-psq/src/handshake/builder.rs

```diff
@@ -0,0 +1,151 @@
+use rand::CryptoRng;
+
+use super::{
+    dhkem::{DHKeyPair, DHPublicKey},
+    initiator::{query::QueryInitiator, registration::RegistrationInitiator},
+    pqkem::{PQKeyPair, PQPublicKey},
+    responder::Responder,
+    HandshakeError as Error,
+};
+
+pub struct Builder<'a, Rng: CryptoRng> {
+    rng: Rng,
+    context: &'a [u8],
+    inner_aad: &'a [u8],
+    outer_aad: &'a [u8],
+    longterm_ecdh_keys: Option<&'a DHKeyPair>,
+    longterm_pq_keys: Option<PQKeyPair<'a>>,
+    peer_longterm_ecdh_pk: Option<&'a DHPublicKey>,
+    peer_longterm_pq_pk: Option<PQPublicKey<'a>>,
+    recent_keys_upper_bound: Option<usize>,
+}
+
+impl<'a, Rng: CryptoRng> Builder<'a, Rng> {
+    /// Create a new builder.
+    pub fn new(rng: Rng) -> Self {
+        Self {
+            rng,
+            context: &[],
+            inner_aad: &[],
+            outer_aad: &[],
+            longterm_ecdh_keys: None,
+            longterm_pq_keys: None,
+            peer_longterm_ecdh_pk: None,
+            peer_longterm_pq_pk: None,
+            recent_keys_upper_bound: None,
+        }
+    }
+
+    // properties
+
+    /// Set the context.
+    pub fn context(mut self, context: &'a [u8]) -> Self {
+        self.context = context;
+        self
+    }
+
+    /// Set the inner AAD.
+    pub fn inner_aad(mut self, inner_aad: &'a [u8]) -> Self {
+        self.inner_aad = inner_aad;
+        self
+    }
+
+    /// Set the outer AAD.
+    pub fn outer_aad(mut self, outer_aad: &'a [u8]) -> Self {
+        self.outer_aad = outer_aad;
+        self
+    }
+
+    /// Set the long-term ECDH key pair.
+    pub fn longterm_ecdh_keys(mut self, longterm_ecdh_keys: &'a DHKeyPair) -> Self {
+        self.longterm_ecdh_keys = Some(longterm_ecdh_keys);
+        self
+    }
+
+    /// Set the long-term PQ key pair.
+    pub fn longterm_pq_keys(mut self, longterm_pq_keys: impl Into<PQKeyPair<'a>>) -> Self {
```

Comment:
We should make the API consistent. The DH above gets just a reference. We should pick a style and stick to it for both types of keys.